rpm/src/main/java/org/eclipse/packager/rpm/signature/PgpHeaderSignatureProcessor.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.packager.rpm.signature;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+import org.bouncycastle.openpgp.PGPSecretKeyRing;
+import org.eclipse.packager.rpm.RpmSignatureTag;
+import org.eclipse.packager.rpm.header.Header;
+import org.pgpainless.PGPainless;
+import org.pgpainless.algorithm.DocumentSignatureType;
+import org.pgpainless.encryption_signing.EncryptionResult;
+import org.pgpainless.encryption_signing.EncryptionStream;
+import org.pgpainless.encryption_signing.ProducerOptions;
+import org.pgpainless.encryption_signing.SigningOptions;
+import org.pgpainless.key.protection.SecretKeyRingProtector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An RSA signature processor for the header section only.
+ */
+public class PgpHeaderSignatureProcessor implements SignatureProcessor {
```

Comment:
Sorry, I am a bit confused, as the RPM header also supports `DSAHEADER`, but I did not actually find anywhere that they key was restricted to RSA as opposed to DSA.

Reply:
I guess I just copied this comment from https://github.com/eclipse/packager/blob/68f8658bf51147ba95c171155880ce4d1d946dcd/rpm/src/main/java/org/eclipse/packager/rpm/signature/RsaHeaderSignatureProcessor.java#L30-L32

Afaict, from the OpenPGP side there is no restriction to any particular algorithm.

Reply:
Yes, this was something in the original code, but based on how RPM does it, I think we should have something like this pseudocode.

```java
// Get the pubkeyAlgo used
pubKeyAlgo = ...
if (pubkeyAlgo == DSA || pubKeyAlgo == EDDSA) {
   signature.putBlob(RpmSignatureTag.DSAHEADER, this.value);
} else if (pubkeyAlgo == RSA) {
     signature.putBlob(RpmSignatureTag.RSAHEADER, this.value);
else {
    throw new RuntimeException("No other algs supported for some reason");
}
```

Reply:
I added a switch statement :)